packages/debug-utils/index.js

```diff
@@ -532,16 +535,19 @@ var DebugUtils = {
   },
 
   formatCurrentInstruction: function (instruction) {
+    // return warning message if the debugger does not have the code for this contract
+    if (!instruction || !instruction.pc) {
```

Comment:
The value `instruction.pc` is a number; it may be equal to 0.  You shouldn't use `!value` to test for the absence of a value when that value is a number, because this will treat numbers with a value of zero as if they are undefined.

With this code, if you type `p` at the beginning of a transaction, you will see this warning, regardless of whether it is appropriate.

Reply:
Actually, sorry, one other comment: Is this really the appropriate place for this check?  This is where the error occurs, yes; but, like, putting the check here is essentially making the assumption that nothing *else* in instruction printing will error.  It *doesn't*, but that's sort of, like, good luck, right?  I don't know if it's a good idea to rely on that; wouldn't it be better to perform the check before any of this stuff is printed?

Reply:
I put the check in index.js since this is the final place so that we only need to check in one place.
Another consideration is to check `instruction` and `instruction.pc` before calling formatCurrentInstruction() and  prints warning message.  Then shall the gas information be shown?
I am OK with either.

Reply:
My point isn't about whether it's called inside `formatCurrentInstruction` or outside `formatCurrentInstruction`, but rather `formatCurrentInstruction` is the correct code to guard in the first place.  Like, my suspicion is that the correct code to guard is the entire segment that prints out the instructions; this currently has no name but it's lines 228-261 in `printInstruction`.  It's true that `formatCurrentInstruction` is the only part in there that currently errors, I'm just not sure it's a good idea to rely on the rest not erroring.

Like, the code structure should ideally reflect the overall logical structure, right?  So if we have three operations `doA()`, `doB()`, and `doC()`, and these operations only make sense when `stuffIsValid` is true, then even if technically `doB()` is the only one that actually *errors* otherwise, surely it makes more sense to do `if (stuffIsValid) { doA(); doB(); doC(); }` rather than `do A(); if (stuffIsValid) do B(); doC(); }`, right?  Because the latter is doing things that are meaningless, even they happen to work out OK.  That's how I see it, at least.

Reply:
Make sense, and calling `formatCurrentInstruction` is the first place to guard.  I look the entire `printInstruction` and do not see any thing else could cause error.  I will move the guard to printer.js.